src/components/CalendarLayout/_components/Calendar/Calendar.tsx

```diff
@@ -1,68 +1,134 @@
-		setSelectedAppointment(appointment)
-		setIsModalVisible(true)
+	const filterAppointments = (value: Dayjs) => {
+		return appointments.filter((appointment) =>
+			dayjs(appointment.planned_date).isSame(value, 'day'),
+		)
 	}
 
 	const dateCellRender = (value: Dayjs) => {
-		const listData = getListData(value)
+		const filteredAppointments = filterAppointments(value).sort(
+			(a, b) => dayjs(a.planned_date).valueOf() - dayjs(b.planned_date).valueOf(),
+		)
+		const filteredEvents = filterEvents(value).sort(
+			(a, b) => dayjs(a.event_date).valueOf() - dayjs(b.event_date).valueOf(),
+		)
+		if (filteredAppointments.length === 0 && filteredEvents.length === 0) {
+			return null
+		}
+
 		return (
-			<ul className="events">
-				{listData.map((item) => (
-					<li key={item.name} onClick={() => showAppointmentDetails(item)}>
-						<Badge status={item.type as BadgeProps['status']} text={item.name} />
-					</li>
-				))}
-			</ul>
+			<div>
+				<Badge
+					count={filteredAppointments.length}
+					style={{ backgroundColor: 'orange', marginRight: '5px' }}
+				/>
+				<Badge
+					count={filteredEvents.length}
+					style={{ backgroundColor: 'blue', marginRight: '5px' }}
+				/>
+				<div>
+					<List
+						dataSource={filteredAppointments}
+						renderItem={(item) => (
+							<List.Item key={item.school_name}>
+								<Badge dot style={{ backgroundColor: 'orange', marginRight: '10px' }} />
+								<span style={{ fontWeight: 'bold', color: '#4a4a4a' }}>{item.school_name}</span>
+								<div style={{ color: '#4a4a4a' }}>{dayjs(item.planned_date).format('HH:mm')}</div>
+							</List.Item>
+						)}
+					/>
+					<List
+						dataSource={filteredEvents}
+						renderItem={(item) => (
+							<List.Item key={item.event_title}>
+								<Badge dot style={{ backgroundColor: 'blue', marginRight: '10px' }} />
+								<span style={{ fontWeight: 'bold', color: '#4a4a4a' }}>{item.event_title} </span>
+								<div style={{ color: '#4a4a4a' }}>{dayjs(item.event_date).format('HH:mm')}</div>
+							</List.Item>
+						)}
+					/>
+				</div>
+			</div>
 		)
 	}
 
-	const handlePrevMonth = useCallback(() => {
-		setCurrentDate((currentDate) => currentDate.subtract(1, 'month'))
-	}, [])
-
-	const handleNextMonth = useCallback(() => {
-		setCurrentDate((currentDate) => currentDate.add(1, 'month'))
-	}, [])
+	const updateCurrentDate = (
+		operation: 'subtract' | 'add',
+		value: number,
+		unit: 'month' | 'year',
+	) => {
+		setCurrentDate((currentDate) => currentDate[operation](value, unit))
+	}
 
+	const monthOptions = useMemo(() => {
+		const options = []
+		const current = currentDate.clone()
+		const months = []
+		for (let i = 0; i < 12; i++) {
+			months.push(current.locale('fr').month(i).format('MMM'))
+		}
+		for (let index = 0; index < 12; index++) {
+			options.push(
+				<Select.Option className="month-item" key={`${index}`}>
+					{months[index]}
+				</Select.Option>,
+			)
+		}
+		return options
```

Comment:
Je comprends l'intérêt de cette fonction mais tu te compliques la vie inutilement. Tu peux simplifier comme ça :
```suggestion
	const monthOptions = useMemo(() => {
		const options = []
		
		for (let i = 0; i < 12; i++) {
		         const month = currentDate.locale('fr').month(i).format('MMM')
		         
		         options.push(
				<Select.Option className="month-item" key={month}>
					{month}
				</Select.Option>,
			)
		}

		return options
```
N'utilise pas l'index comme key, React n'aime pas ça.

Je te conseille de lire la [doc antd](https://ant.design/components/select#usage-upgrade-after-5110) avant d'utiliser les composants. Il est indiqué dans la doc du composant `Select` que `Select.Option` est déprécié.
<img width="640" alt="Screenshot 2024-04-10 at 21 11 15" src="https://github.com/SchoolEvent/schoolevent-client/assets/76203025/48d373ff-d111-4d69-b3c3-b7006d46e7b4">